### DISCUSSION TITLE
how to know current page dimensions

### DISCUSSION THREAD

**Discussion by abubelinha** [py-pdf/fpdf2#981] [Q&A] [answered]
I want to define some functions for drawing stuff in current page, just using pdf.line() to loop a list of points with x,y coordinates.
My designs need to be proportional to page size (so I will recalculate coordinates according to that).

Is it possible that I can get page dimensions from inside my function? 
(i.e. `pdf.get_page_dimensions()`, or something like that.

I couldn't find such feature. If it does not exist, I would need to know dimensions in advance and pass them to my function (but I'd prefer not having to be passing around page dimensions, if possible).

```Python
def mydrawing(points=[])
    # code to know current page width and page height?
    # pw & ph would be     
    for p in points:
        # code to calculate max and min x,y values of all points in list
    for p in points:
        # point coordinates will be redefined based on max & min x,y values, 
        # so my drawing will be proportional to current page size
        self.line(... whatever ...)

points = [[30,40], [50,60], ...]
pdf.mydrawing(points)
```

Thanks
@abubelinha

**Comment by andersonhc** [comment, marked as answer]
fpdf.w = Page width
fpdf.h = Page height
fpdf.epw = Effective page width: the page width minus its horizontal margins.
fpdf.eph = Effective page height: the page height minus its vertical margins.

**Reply by abubelinha** [reply]
Thanks a lot. 

Is this documented somewhere? 
I searched for epw in [fpdf2 docs](https://py-pdf.github.io/fpdf2/) but got no results.
Also tried old [fpdf docs](https://pyfpdf.readthedocs.io/en/latest/search.html?q=epw)

**Reply by andersonhc** [reply]
I don't think it's documented.
Do you want to do it?
I guess [Margins](https://github.com/py-pdf/fpdf2/blob/master/docs/Margins.md) would be a good place - it only shows how to set them but not how to get.

**Reply by abubelinha** [reply]
I am not a git user. I am only used to github web interface.

Do you mean editing it like this? 
https://github.com/py-pdf/fpdf2/compare/master...abubelinha:fpdf2:patch-1

I tried to put some links I found for [`fpdf.epw`](https://py-pdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.epw) and [`fpdf.eph`](https://py-pdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.eph) (couldn't find the same for `fpdf.w` and `fpdf.h`)
What should I do next? Should I click "create pull request"?

**Reply by Lucas-C** [reply]
I think we (`fpdf2` maintainers) are mostly `git` command line users, or even users of [IDEs](https://en.wikipedia.org/wiki/Integrated_development_environment) with plugins, so we are not used to create PRs through the GitHub web interface 😅

But that's totally doable!

I found some documentation about this there:
* https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request
* https://pixelpioneers.co/blog/2017/using-github-without-the-command-line

And a contribution to better document `.w` / `.h` / `.epw` / `.eph` would be totally welcome!
Where do you plan to mention them?
I think we just have basic docstrings for `.epw` & `.eph` those:
* https://py-pdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.eph
* https://py-pdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.epw
